### PR TITLE
Remove persistence of PeerStore to DB

### PR DESCRIPTION
**Pull request by saketh-are**
After the changes in https://github.com/near/nearcore/issues/8579:
- Connected nodes are persisted to DB via the ConnectionStore
- Connections to those nodes are re-established upon restarting the node

Previously, we persisted the PeerStore to disk to support faster network bootstrapping after restarting a node. Now that the ConnectionStore serves that purpose, it is no longer necessary to persist the PeerStore. Unlike ConnectionStore, PeerStore contains unverified information and may contain spam shared by malicious nodes. This PR both removes unnecessary load on the DB and limits persistence of malicious data.

This PR
- Removes all interaction between the PeerStore and the DB
- Updates APIs of the PeerStore accordingly as it no longer produces DB read/write errors
- Updates PeerStore tests accordingly

### CODE REVIEW

File: chain/network/src/peer_manager/peer_store/mod.rs [hunk 1]
```diff
@@ -169,7 +144,7 @@ impl Inner {
                 }
             }
         }
-        Ok(self.store.delete_peer_states(peer_ids)?)
+        Ok(())
```
Comment (pompon0) [inline chain/network/src/peer_manager/peer_store/mod.rs:147]: there is no error conditions, so  anyhow::Result is not needed.

File: chain/network/src/peer_manager/peer_store/tests.rs [hunk 2]
```diff
@@ -371,191 +286,70 @@ fn check_add_peers_overriding() {
...
-    // Check we are able to recover from store previous signed connection
-    let peer_store_2 =
-        PeerStore::new(&clock.clock(), make_config(&[], Default::default(), false), store).unwrap();
-    assert!(check_exist(&peer_store_2, &peers_id[0], Some((addrs[0], TrustLevel::Indirect))));
-    assert!(check_integrity(&peer_store_2));
-}
-
-#[test]
-fn check_ignore_blacklisted_peers() {
-    let clock = time::FakeClock::default();
```
Comment (pompon0) [inline chain/network/src/peer_manager/peer_store/tests.rs:448]: do we have other tests checking that blacklist actually works?

**Review by pompon0 — APPROVED** [review]
LGTM!

File: chain/network/src/peer_manager/peer_store/mod.rs [hunk 3]
```diff
@@ -1,19 +1,16 @@
...
 use near_primitives::network::PeerId;
-use near_store::db::Database;
 use parking_lot::Mutex;
 use rand::seq::IteratorRandom;
 use rand::thread_rng;
 use std::net::SocketAddr;
 use std::ops::Not;
-use std::sync::Arc;
 
 #[cfg(test)]
```
Comment (mm-near) [inline chain/network/src/peer_manager/peer_store/mod.rs:15]: let's add a high level comment on what 'PeerStore' role is right now.

As AFAIK it only holds the stuff in memory now, right? so we should also mention what happens on restart.

File: chain/network/src/store/schema/mod.rs [hunk 4]
```diff
@@ -170,13 +96,6 @@ impl Column for AccountAnnouncements {
...
 }
 
-pub(super) struct Peers;
-impl Column for Peers {
-    const COL: DBCol = DBCol::Peers;
-    type Key = Borsh<PeerId>;
-    type Value = KnownPeerStateRepr;
-}
-
 pub(super) struct RecentOutboundConnections;
```
Comment (mm-near) [inline chain/network/src/store/schema/mod.rs:99]: we should also update the comment in columns.rs (about the DBCol::Peers).

**Review by mm-near — APPROVED** [review]
LGTM - please also update the CHANGELOG.md

**Pull request merged**